shared-libs/transitions/src/transitions/index.js

```diff
@@ -0,0 +1,331 @@
+    if (info.transitions && info.transitions[key]) {
+      return parseInt(doc._rev) === parseInt(info.transitions[key].last_rev);
+    }
+    if (doc.transitions && doc.transitions[key]) {
+      return parseInt(doc._rev) === parseInt(doc.transitions[key].last_rev);
+    }
+    logger.debug(
+      `isRevSame tested true on transition ${key} for doc ${change.id} seq ${change.seq}`
+    );
+    return false;
+  };
+
+  /*
+   * Ignore deleted records, confirm transition has filter function and
+   * skip transition if filter returns false.
+   *
+   * If transition was already applied then only run again if doc rev is not
+   * equal to transition rev.  If revs are the same then transition just ran.
+   */
+  return Boolean(
+    change &&
+      !change.deleted &&
+      doc &&
+      !isRevSame(doc, info) &&
+      transition.filter(doc, info)
+  );
+};
+
+/*
+ * All transitions reference the same change.doc and work in series to
+ * apply changes to it.  A transition is free to make async calls but the next
+ * transition will only run after the previous transitions's callback is
+ * called.  This is a performance optimization that allows us to apply N
+ * transitions (updates) to a document with the cost of a single database
+ * change/write.
+ */
+const applyTransition = ({ key, change, transition }, callback) => {
+  if (!canRun({ key, change, transition })) {
+    logger.debug(
+      `canRun test failed on transition ${transition.key} for doc ${change.id} seq ${change.seq}`
+    );
+    return callback();
+  }
+
+  logger.debug(
+    `calling transition.onMatch for doc ${change.id} and transition ${key} seq ${change.seq}`
+  );
+
+  /*
+   * Transitions are async and should return true if the document has
+   * changed.  If a transition errors then log the error, but don't return it
+   * because that will stop the series and the other transitions won't run.
+   */
+  transition
+    .onMatch(change)
+    .then(changed => {
+      logger.debug(
+        `finished transition ${key} doc ${change.id} is ${changed ? 'changed' : 'unchanged'} seq ${change.seq}`
+      );
+      if (!changed) {
+        return changed;
+      }
+      return infodoc.updateTransition(change, key, true).then(() => {
+        return changed;
+      });
+    })
+    .catch(err => {
+      // adds an error to the doc but it will only get saved if there are
+      // other changes too.
+      const message = err.message || JSON.stringify(err);
+      utils.addError(change.doc, {
+        code: `${key}_error'`,
+        message: `Transition error on ${key}: ${message}`,
+      });
+      logger.error(`transition ${key} errored on doc ${change.id} seq ${change.seq}: ${JSON.stringify(err)}`);
+      if (!err.changed) {
+        return false;
+      }
+      return infodoc.updateTransition(change, key, false).then(() => {
+        return true;
+      });
+    })
+    .then(changed => callback(null, changed)); // return the promise instead
+};
+
```

Comment:
processChange is stripped of everything that is Sentinel related. It only hydrates the doc, creates/updates infodocs and applies loaded transitions.